uk.co.compucorp.civicrm.tasksassignments/js/test/filters/due_test.js

```diff
@@ -135,6 +278,101 @@ define([
       it('hides the result for "dateRange" in the past', function() {
         expect($filter('filterByDue')(mock, 'dateRange', pastDateRange).length).toBe(0);
       });
+
+      it('shows the result for "dateRange" if from and until dates are empty', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyDateRange).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is in the past', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyFromDateAndUntilDateInPast).length).toBe(0);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is today', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyFromDateAndUntilDateIsToday).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if from date is empty and until date is in the future', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyFromDateAndUntilDateInTheFuture).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is in the past', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyUntilDateAndFromDateInThePast).length).toBe(1);
+      });
+
+      it('shows the result for "dateRange" if until date is empty and from date is today', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyUntilDateAndFromDateIsToday).length).toBe(1);
+      });
+
+
+      it('shows the result for "dateRange" if until date is empty and from date is in the future', function() {
+        expect($filter('filterByDue')(mock, 'dateRange', emptyUntilDateAndFromDateInTheFuture).length).toBe(0);
+      });
+    });
+
+    describe('due date is empty', function() {
+      var mock;
+      beforeEach(function() {
+        mock = [{
+          activity_date_time: null
+        }];
+      });
+
+      it('hides the result for "overdue"', function() {
```

Comment:
As I said in a previous comment , this doesn't represent true of false rather than the number of results return.